### PR TITLE
SCRUM-850 AZ-agnostic EC2 launch

### PR DESCRIPTION
**Pull request by mluypaert**
EC2 instance launching through ansible is occasionally fails due to `InsufficientInstanceCapacity` errors. This PR should fix that (at least for the developers repo, I'll replicate them to the devOps repo once approved), by trying to launch an instance in all availability zones (capacity error are typically limited to a single or a few availability zones) and with different configurations (spot vs on-demand, in case all spot capacity for an instance type would be depleted), until one launch succeeds. Code for using spot capacity has also been fixed, as it was defunct (fix is depending on alliance-genome/agr_base_linux_env#8 though for correct functionality).

### CODE REVIEW

File: tasks/check_ssh.yml [hunk 1]
```diff
@@ -1,11 +0,0 @@
...
-  wait_for:
-    host: "{{ item.public_dns_name }}"
-    port: 22
-    delay: 5
-    timeout: 300
-    state: started
-  with_items: "{{ ec2.instances }}"
-
-- pause:
-    seconds: 15
```
Comment (oblodgett) [inline tasks/check_ssh.yml:11]: Curious why the removal of check ssh?

Comment (christabone) [reply tasks/check_ssh.yml:11]: @oblodgett It looks like it's moved to another place in the code, if I'm not mistaken?

https://github.com/alliance-genome/agr_ansible_developers/pull/13/files#diff-a385ec69082237ced4df664197bcaeb091c951cd67924eb59777cb19a9cda7d6R99

**Comment by christabone** [comment]
This all looks great -- I don't see any glaring issues. I'm assuming you've had a chance to launch it and test it out? Making a guess based on all those AWS channel messages from the other day. :)

**Review by christabone — APPROVED** [review]

**Comment by christabone** [comment]
I approve it but maybe just double check that I'm correct in my reply to Olin before you merge? Is there anything else you want to mention @oblodgett?

File: tasks/check_ssh.yml (hunk 1 above)
Comment (mluypaert) [reply tasks/check_ssh.yml:11]: @christabone @oblodgett Yes, rest assured that the ssh check is still there (excactly where @christabone linked to). I simply removed the subtask file and put the content (the ssh check task) straight in the playbook, as it is not being used anywhere else but that playbook (and only once) and thus having it in a separate task file was only making the flow harder to follow.

**Comment by mluypaert** [comment]
> This all looks great -- I don't see any glaring issues. I'm assuming you've had a chance to launch it and test it out? Making a guess based on all those AWS channel messages from the other day. :)

Haha yes, I've been testing a few versions of this code over and over until one (this PR) actually worked. I didn't realise I was spamming the AWS channel so hard 😂

**Pull request merged**